keras_core/layers/preprocessing/discretization.py

```diff
@@ -241,20 +237,15 @@ def call(self, inputs):
         if not isinstance(
             inputs,
             (
-                tf.Tensor,
-                tf.SparseTensor,
-                tf.RaggedTensor,
                 np.ndarray,
                 backend.KerasTensor,
             ),
         ):
-            inputs = tf.convert_to_tensor(
-                backend.convert_to_numpy(inputs), dtype=self.input_dtype
+            inputs = backend.convert_to_tensor(
```

Comment:
We could try `self._convert_input_args = True` and remove this

Reply:
done